plugins/inputs/redfish/redfish.go

```diff
@@ -104,138 +93,87 @@ type Placement struct {
 	Rack string `json:"Rack"`
 	Row  string `json:"Row"`
 }
-
-var h Hostname
-var t Temperatures
-var f Fans
-var p PowerSupplies
-var v Voltages
-var php PowerSupplieshp
-var l Location
-
-func basicAuth(username, password string) string {
-	auth := username + ":" + password
-	return base64.StdEncoding.EncodeToString([]byte(auth))
+type Redfish struct {
+	Host              string `toml:"host"`
+	BasicAuthUsername string `toml:"basicauthusername"`
+	BasicAuthPassword string `toml:"basicauthpassword"`
+	Id                string `toml:"id"`
+	Server            string `toml:"server"`
+	client            http.Client
+	tls.ClientConfig
+	Timeout     internal.Duration `toml:"timeout"`
+	hostname    Hostname
+	temperature Temperatures
+	fan         Fans
+	powerdell   PowerSupplies
+	voltage     Voltages
+	powerhp     PowerSupplieshp
+	location    Location
 }
 
-func getThermal(host, username, password, id string) error {
-	r := fmt.Sprint(host, "/redfish/v1/Chassis/", id, "/Thermal")
-	client := &http.Client{}
-	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
-	req, err := http.NewRequest("GET", r, nil)
-	req.Header.Add("Authorization", "Basic "+basicAuth(username, password))
-	req.Header.Set("Accept", "application/json")
-	req.Header.Set("Content-Type", "application/json")
-	resp, err := client.Do(req)
-	if err != nil {
-		return err
+func (r *Redfish) getMetrics() error {
+	url := make(map[string]map[string]interface{})
```

Comment:
This is alright, but the arbitrary structure of maps is a bit unusual. Typically you'd use field tags to specify the json names and unmarshal into a struct instance all in one go.

eg... 
```
type payload struct {
  ...
  Fan Fans `json:"fanpointer"
  ...
}
// and in the getMetrics function
p := &payload{}
json.Unmarshal(body, p)
```
You have some flexibility to load everything at once this way, and can customize struct-specific decoding logic with  `UnmarshalJSON` [example](https://golang.org/pkg/encoding/json/#example__customMarshalJSON).

If you're going to go that route, I'd probably separate the expected payload structure from the Redfish struct.

That might make it a little more clear about what to do with the "endpoint" key, which isn't really a result value, it's just being stored in the structure for easy access while iterating over resources you want to query. This map is doing a lot!